✨   MS preflight checks to improve cluster stability (#8595)
kubernetes-sigs/cluster-api: docs/book/src/reference/labels_and_annotations.md

| cluster.x-k8s.io/replicas-managed-by                              | It can be applied to MachinePool resources to signify that some external system is managing infrastructure scaling for that pool. See [the MachinePool documentation](../developer/architecture/controllers/machine-pool.md#externally-managed-autoscaler) for more details.                                                                                                                                                                                                                                                                                |

> | cluster.x-k8s.io/skip-machineset-preflight-checks                 | It can be applied on MachineSet resources to specify the list of pre-flight checks to skip during MachineSet reconciliation. The value is a coma seperated list of skippable pre-flight checks. Supported pre-flight checks are: All, KubeadmVersion, KubernetesVersion, ControlPlaneStable.                                                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| cluster.x-k8s.io/skip-machineset-preflight-checks                 | It can be applied on MachineDeployment and MachineSet resources to specify a comma-separated list of preflight checks that should be skipped during MachineSet reconciliation. Supported pre-flight checks are: All, KubeadmVersion, KubernetesVersion, ControlPlaneStable.                                                                                                                                                                                                                                                                |
```
